corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -37,23 +49,60 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
 
         ui: {
             submitButton: '#query-submit-button',
+            valueDropdown: 'select.query-field',
         },
 
         events: {
+            'change @ui.valueDropdown': 'changeDropdown',
             'click @ui.submitButton': 'submitAction',
         },
 
-        submitAction: function (e) {
-            e.preventDefault();
-            var payload = {};
-            var fields = $(".query-field");
-            var model = this.parentModel;
-            fields.each(function (index) {
+        getAnswers: function () {
+            var $fields = $(".query-field"),
+                answers = {},
+                model = this.parentModel;
+            $fields.each(function (index) {
                 if (this.value !== '') {
-                    payload[model[index].get('id')] = this.value;
+                    answers[model[index].get('id')] = this.value;
```

Comment:
For `select` fields, would `value` here be the index of the selected choice ?

Reply:
It should be, yes.